### PR TITLE
Feat/appearance settings

### PR DESCRIPTION
**Pull request by daianaszwimer**
On the blog cards layout I used `line-clamp` which is not compatible with IE yet. I've tested it and it looks okay I think (it doesn't break the functionality and the page doesn't break)
https://caniuse.com/?search=line-clamp
I attached the comparison, on my local machine `Chrome on Mac OS` (1) vs `IE on Windows 10` (2)

(1)
image

(2)
image

### CODE REVIEW

File: db/migrate/20210326165901_add_appearance_settings_to_authors.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class AddAppearanceSettingsToAuthors < ActiveRecord::Migration[5.0]
+  def change
+      add_column :authors, :cover_style, :string, default: 'full'
```
Comment (atmoio) [inline db/migrate/20210326165901_add_appearance_settings_to_authors.rb:3]: I'm a bit concerned about the migration for the `default` value taking a long time. @karolsojko do you think this is fine or do we want to be careful about this?

File: db/schema.rb [hunk 2]
```diff
@@ -10,15 +10,15 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 20210326232803) do
+ActiveRecord::Schema.define(version: 20210326223431) do
 
-  create_table "authors", force: :cascade, options: "ENGINE=InnoDB DEFAULT CHARSET=latin1" do |t|
+  create_table "authors", force: :cascade, options: "ENGINE=InnoDB DEFAULT CHARSET=utf8" do |t|
```
Comment (daianaszwimer) [inline db/schema.rb:15]: Oh, I didn't realize my local DB had a different charset, should I update it and run the migrations again to re-generate this file and avoid the unnecessary diff? @mobitar

File: client/app/components/shared/RadioButton.jsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+}) => {
+    const handleChange = (event) => {
+        onChange(event.target.value);
+    };
+    const handleSVGClick = () => {
+        onChange(value);
+    };
+
+    return (
+        <div className="form-section radio_button__container">
```
Comment (antsgar) [inline client/app/components/shared/RadioButton.jsx]: To make these classes 100% BEM, the block's name should be separated by hyphens (-), for example `radio-button__container`

File: client/app/components/shared/RadioButtonGroup.jsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+const RadioButtonGroup = ({
+    options, onChange, name, selected, text,
+}) => {
+    const handleChange = (value) => {
+        onChange(name, value);
+    };
+    return (
+        <div className="form-section radio_button_group__container">
+            <p className="label p2">{text}</p>
+            <div className="form-section radio_button_group__options-container">
```
Comment (antsgar) [inline client/app/components/shared/RadioButtonGroup.jsx]: Same here, this one should be `radio-button-group__options-container` to make it 100% BEM 🙂

File: client/app/assets/icons/index.js [hunk 5]
```diff
@@ -17,6 +17,9 @@ import IcListed from "./ic-listed.svg";
 import IcMenu from "./ic-menu.svg";
 import IcMoreHorizontal from "./ic-more-horizontal.svg";
 import IcOpenIn from "./ic-open-in.svg";
+import IcPalette from "./ic-pallete.svg";
```
Comment (antsgar) [inline client/app/assets/icons/index.js]: nit: ic-palette typo

File: client/app/components/authors/settings/Appearance.jsx [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+
+    return (
+        <form onSubmit={handleSubmit}>
+            <RadioButtonGroup
+                id="author-cover-style"
+                onChange={handleAuthorChange}
+                selected={editedAuthor.cover_style}
+                text="Cover Style"
+                name="cover_style"
+                options={getOptions("coverStyle")}
```
Comment (antsgar) [inline client/app/components/authors/settings/Appearance.jsx]: We could probably extract "coverStyle" and "blogLayoutStyle" strings to constants for better readability

File: client/app/components/shared/RadioButton.scss [hunk 7]
```diff
@@ -0,0 +1,38 @@
+.form-section.radio_button__container {
+  border: 1px solid $color-black-opacity-16;
+  border-radius: 4px;
```
Comment (antsgar) [inline client/app/components/shared/RadioButton.scss]:
```suggestion
  border-radius: $border-radius-4;
```

File: client/app/components/shared/RadioButton.scss [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+.radio_button__icon-container {
+  margin-right: $spacing-12;
+  height: 20px;
+  width: 20px;
+}
+
+.radio_button__input-container {
+  display: flex;
+  padding: 12px;
```
Comment (antsgar) [inline client/app/components/shared/RadioButton.scss]:
```suggestion
  padding: $spacing-12;
```

File: client/app/components/shared/RadioButtonGroup.scss [hunk 9]
```diff
@@ -0,0 +1,4 @@
+.form-section.radio_button_group__options-container {
+  flex-direction: row;
+  padding-top: 6px;
```
Comment (antsgar) [inline client/app/components/shared/RadioButtonGroup.scss]:
```suggestion
  padding-top: $spacing-6;
```

File: db/schema.rb (hunk 2 above)
Comment (atmoio) [reply db/schema.rb:15]: UTF8 is the better encoding here anyway so I think it's fine 👍

File: config/routes.rb [hunk 10]
```diff
@@ -14,6 +14,7 @@
       post 'domain_request'
       post 'delete_domain'
       post 'delete_all_data'
+      put 'appearance' => 'authors#update_appearance'
```
Comment (atmoio) [inline config/routes.rb]: I don't think this should be it's own endpoint. Should just be part of Authors#Update.

**Review by antsgar — APPROVED** [review]

File: app/controllers/authors_controller.rb [hunk 11]
```diff
@@ -65,6 +65,8 @@ def show
       if all_posts.count > POST_LIMIT && all_posts.last.created_at < @posts.last.created_at
         @posts.last.created_at.to_i
       end
+    @should_show_condensed_cover = @display_author.cover_style == "condensed"
```
Comment (atmoio) [inline app/controllers/authors_controller.rb]: "condensed" and other options should be constants somewhere and not magic strings whose values are not predefined.

File: config/routes.rb (hunk 10 above)
Comment (daianaszwimer) [reply config/routes.rb]: @mobitar I tried with that approach at first and we have two options there:
- We always send all the author attributes (on the "General" section we send the "Appearance" data and viceversa)
- On the `update` method we check the params and if they're `nil` we don't update them

Which one do you think is the best?

File: client/app/components/shared/RadioButton.jsx (hunk 3 above)
Comment (daianaszwimer) [reply client/app/components/shared/RadioButton.jsx]: oops! Nice catch, I just pushed the fix @antsgar

File: client/app/components/shared/RadioButtonGroup.jsx (hunk 4 above)
Comment (daianaszwimer) [reply client/app/components/shared/RadioButtonGroup.jsx]: Fixed! @antsgar

File: client/app/components/authors/settings/Appearance.jsx (hunk 6 above)
Comment (daianaszwimer) [reply client/app/components/authors/settings/Appearance.jsx]: yes! you're correct, fixed @antsgar

File: client/app/assets/icons/index.js (hunk 5 above)
Comment (daianaszwimer) [reply client/app/assets/icons/index.js]: fixed! @antsgar

File: config/routes.rb (hunk 10 above)
Comment (atmoio) [reply config/routes.rb]: I think sending all attributes makes the most sense. I would also research into the most idiomatic way to do this with Rails. I believe it is the first option.

Comment (daianaszwimer) [reply config/routes.rb]: done! @mobitar

File: app/controllers/authors_controller.rb (hunk 11 above)
Comment (daianaszwimer) [reply app/controllers/authors_controller.rb]: agree, fixed! @mobitar

**Review by atmoio — APPROVED** [review]
